crates/hulk_behavior_simulator/src/autoref.rs

```diff
@@ -86,6 +89,10 @@ fn autoref(
             if ball.state.is_none() {
                 ball.state = Some(SimulatorBallState::default());
             };
+            if !state.is_whistling {
+                game_controller_commands.send(GameControllerCommand::Whistle);
+                state.is_whistling = true;
+            }
```

Comment:
Suggestion: Make a new resource for the whistle which you can modify here and access in the `cycle_robots` system to update the `FilteredWhistle` fake data.
Then neither the autoref nor the  `GameControllerControllerState` resource needs to store this flag.

Also, using a Timer which automatically expires. That way we get to whistle for a second or so instead of just one cycle.
As long as the timer hasn't expired yet, set `is_detected` to true and update the `last_detection` time.

Reply:
Implemented like suggested. The problem still exists that I need the whistle to only whistle once for a second but I want the game controller delay to be 15. For this the gamecontroller uses the `has_whistled` whereas the robots use `is_whistling`. If you have suggestions for better names for these variables or how I can better handle these two things let me know.